src/parsers/perf/perfparser.cpp

```diff
@@ -1614,7 +1614,7 @@ void PerfParser::startParseFile(const QString& path)
 
         // note: file is always readable and in supported format here,
         //        already validated in initParserArgs()
-        QFile file(path);
+        QFile file(m_parserArgs[1]);
```

Comment:
`path` is unchanged (we'd need an additional object or return a QString from `initParserArgs()`, the changed version is only placed into the `m_parserArgs[]`.

Please drop a note how you would like this to be tackled and I look into doing that.

But in any case I'm interested why this isn't trhead safe - the code that adjusts `m_parserArgs` is always executed up-front, no?

Reply:
then make it `path = m_parserArgs[1]` in the lambda capture list to fix the thread safety issue.

it is not thread safe because the lambda is executed within the context of another thread, and we might call `initParserArgs` while the background thread executes